R/buffer.R

```diff
@@ -38,6 +49,8 @@ buffer_indices <- function(data, indices, radius, buffer, call = rlang::caller_e
   # only run radius checks if radius is not NULL (to prevent NAs from >)
   run_radius <- !is.null(radius)
   if (run_radius && units::set_units(radius, NULL) > 0) {
+    # In case `buffer` has no units, assume it's in the same units as `data`
+    if (!identical(sf::st_crs(data), sf::NA_crs_)) units(radius) <- units(distmat)
     # In case `radius` has no units, assume it's in the same units as `data`
     units(radius) <- units(distmat)
```

Comment:
this looks like it's not quite what you intended?

Reply:
Gah -- messed up the copy and paste, thanks